### PR TITLE
removed `waveunit` keywords from header where not used for issue 2450

### PR DESCRIPTION
**Pull request by getsanjeev**
removed `waveunit` keywords from header where `wavelnth` keyword not in use
issue #2450

### CODE REVIEW

**Comment by Cadair** [comment]
The changes look good, and but I think you need to merge master into this and make sure the tests pass.

**Comment by getsanjeev** [comment]
@Cadair Okay I will see that. Thanks!

**Comment by wtbarnes** [comment]
@getsanjeev I think there has been a slight miscommunication here. 

Looking at the changeset, it seems you've removed the `waveunit` keyword from several instrument sources. The `waveunit` keyword should ONLY be removed from instrument sources where it does not make sense to include a `wavelnth`. (See my comments).

In PR #2427, you removed a few instances where the `wavelnth` keyword  in the header was being set to Nan because `Nan` is not a valid value according to the FITS standard. `Nan` was (presumably) added as a filler value for instruments where the "wavelength" designation was not appropriate, e.g. the X-Ray Telescope (XRT) on Hinode where the channels are denoted by different filter wheel combinations.

For imaging instruments where the different channels are denoted by the channels on which they are centered (e.g. SOHO EIT, SDO AIA, STEREO EUVI), it is important to keep the `waveunit` keyword as it gives units to the value in the `wavelnth` keyword for that particular measurement.

File: sunpy/map/sources/soho.py [hunk 1]
```diff
@@ -65,8 +65,7 @@ def __init__(self, data, header, **kwargs):
         GenericMap.__init__(self, data, header, **kwargs)
 
         # Fill in some missing info
-        self.meta['detector'] = "EIT"
-        self.meta['waveunit'] = "Angstrom"
```
Comment (wtbarnes) [inline sunpy/map/sources/soho.py]: `waveunit` should NOT be removed here. EIT channels are denoted by wavelengths and thus the `wavlnth` keyword needs a unit attached to it here

File: sunpy/map/sources/stereo.py [hunk 2]
```diff
@@ -35,11 +35,9 @@ class EUVIMap(GenericMap):
     def __init__(self, data, header, **kwargs):
 
         GenericMap.__init__(self, data, header, **kwargs)
-
         self._nickname = "{0}-{1}".format(self.detector, self.observatory[-1])
         self.plot_settings['cmap'] = cm.get_cmap('sohoeit{wl:d}'.format(wl=int(self.wavelength.value)))
-        self.plot_settings['norm'] = ImageNormalize(stretch=source_stretch(self.meta, PowerStretch(0.25)))
-        self.meta['waveunit'] = 'Angstrom'
```
Comment (wtbarnes) [inline sunpy/map/sources/stereo.py]: Same as the above comment about EIT. The `waveunit` keyword should NOT be removed here.

**Comment by nabobalis** [comment]
Would we want a check in Generic Map for this?

**Comment by Cadair** [comment]
@nabobalis a check for what? wavelength but no waveunit?

**Comment by nabobalis** [comment]
That or vice versa.

**Comment by Cadair** [comment]
We *could* put it in the metadata validatior. I am not sure what good it will other than throw a warning though, so I am not really sure it's worth it.

**Comment by nabobalis** [comment]
Fair enough. I withdraw my comments.

**Comment by wafels** [comment]
LASCO has filters with broad bandpasses: see http://adsabs.harvard.edu/cgi-bin/nph-bib_query?1995SoPh..162..357B&db_key=AST . STEREO COR1 and COR2 also have relatively broad bandpasses: https://rd.springer.com/article/10.1007/s11214-008-9341-4

**Pull request merged**